src/types.ts

```diff
@@ -146,6 +130,8 @@ export interface ModuleOptions {
   disableHMR?: boolean
 }
 
+// Hooks TODO: either deprecate or make hooks read-only. Modifications from hooks should rather be done by addition of new configs with defuFn strategy.
```

Comment:
might need to deprecate `loadConfig` and `resolvedConfig` already :/

Reply:
so resolvedConfig could be the first to be deprecated by saying to use `tailwindcss:config` hook and then call resolveConfig yourself, but then there may be changes within `tailwindcss:config` itself that may not be picked within that hook